Added support for instanceMetadataOptions in Ocean AWS and ECS VNG.

sharadkesarwani opened the pull request:
Added support for instanceMetadataOptions in Ocean AWS and ECS VNG.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-13265
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-13358
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-13317

sharadkesarwani committed:
Added support for instanceMetadataOptions in Ocean AWS and ECS VNG.

Added support for instanceMetadataOptions in Ocean AWS and ECS VNG.

sharadkesarwani committed:
Updated document and example

Updated document and example

sharadkesarwani committed:
Handled int default value.

Handled int default value provided by terraform in case field is not set by user.

sharadkesarwani committed:
Upgraded dependencies.

Upgraded dependencies.

sharadkesarwani committed:
fixed dependency upgrade

fixed dependency upgrade

chandra1-n commented on spotinst/ocean_ecs_launch_specification/fields_spotinst_ocean_ecs_launch_specification.go:
```diff
@@ -611,6 +611,7 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
 					string(HTTPPutResponseHopLimit): {
 						Type:     schema.TypeInt,
 						Optional: true,
+						Default:  1357997531,
```
this file doesn't need any change

sharadkesarwani replied:
Made changes in this file to fix existing issue in ECS cluster creation, handled negative value passed by user to throw error and refrain resource update.

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-13265-AWS-EG-Ocean-TF-Set-Access-to-Instance-MetaData-Tags

sharadkesarwani committed:
Fixed existing issue in ocean aws create cluster by handling int default value.

Fixed existing issue in ocean aws create cluster by handling int default value and updated code for Ocean ECS launchspec.

anuragsharma-123 commented on spotinst/ocean_aws_launch_configuration/fields_spotinst_ocean_aws_launch_configuration.go:
```diff
@@ -701,10 +702,12 @@ func expandInstanceMetadataOptions(data interface{}) (*aws.InstanceMetadataOptio
-		instanceMetadataOptions.SetHTTPPutResponseHopLimit(spotinst.Int(v))
-	} else {
-		instanceMetadataOptions.SetHTTPPutResponseHopLimit(nil)
+	if v, ok := m[string(HTTPPutResponseHopLimit)].(int); ok {
```
Can we add a comment why this default value is used?

sharadkesarwani replied:
The default value mentioned is used to set HTTPPutResponseHopLimit field to null when the customer doesn't want to set this param, as terraform set it 0 for integer type param by default.

sharadkesarwani committed:
Added comment for describing usage of dummy default value.

Added comment for describing usage of dummy default value.
